integration-tests/initiateDataRequest.spec.ts

```diff
@@ -24,28 +43,32 @@ describe('Submit a PII request with approved ticket data', () => {
     })
 
     it('Should log a success in cloud watch if Zendesk request is valid', async () => {
-      // Cloudwatch - Fetch latest log stream containing the ticket details
-      const webhookReceivedMessage = 'INFO received Zendesk webhook'
-      const eventLogStream = await waitForLogStreamContainingEvent(
-        webhookReceivedMessage,
-        `"zendeskId`,
-        `"${ticketId}`
-      )
-      const logStreamRequestID = extractRequestIDFromEventMessage(
-        eventLogStream.eventMessage
-      )
-
-      // filter for ticket's successful validation event in the same log stream
-      const validRequestFilterPattern = `"${logStreamRequestID}" Sent data transfer`
-      console.log(`VALIDATION FILTER PATTERN: ${validRequestFilterPattern}`)
-      const validationEvents = await getMatchingLogEvents(
-        validRequestFilterPattern,
-        eventLogStream.logStreamName
-      )
-
-      expect(validationEvents.length).toEqual(1)
-      console.log(`VALIDATION EVENT: ${validationEvents[0].message}`)
-      expect.stringContaining('Sent data transfer queue message with id')
+      const initiateDataRequestEvents =
+        await getCloudWatchLogEventsGroupByMessagePattern(
+          INITIATE_DATA_REQUEST_LAMBDA_LOG_GROUP,
+          WEBHOOK_RECEIVED_MESSAGE,
+          'zendeskId',
+          ticketId
+        )
+
+      expect(
+        isLogPresent(initiateDataRequestEvents, DATA_SENT_TO_QUEUE_MESSAGE)
+      ).toEqual(true)
+
+      const messageId = getQueueMessageId(initiateDataRequestEvents)
+
+      if (messageId) {
```

Comment:
Question if this is a good idea or not. Checking the next lambda involves waiting a fair while as the messave goes to a queue, and that message may not be read instantly.

* Would it be enough just to check that the log message is there with the queue id? that's inferred at the moment
* We could increase the number of tries or waits when searching for the logs? This is currently 30 attempts, perhaps we could give the option to customise this value?

Reply:
Since no one responded, I have now added an attempts parameter, which might be useful regardless. Question still stands about checking the next lambda log group though!

Reply:
We should make sure it reaches the queue to prove out the integration up to that circuit breaker.  What you have now looks good.

Reply:
Since we're going to be checking stuff later down the line, checking the next lambda is useful.